src/viam/sdk/common/kinematics.hpp

```diff
@@ -0,0 +1,49 @@
+#pragma once
+
+#include <boost/variant/variant.hpp>
+#include <vector>
+
+namespace viam {
+namespace sdk {
+
+// Base class for use below in defining kinematics data strong typedefs
+template <class Tag>
+struct raw_bytes {
+    // Pre c++17 our derived classes aren't aggregate initializable so we need to define
+    // and using declare some ctors
+
+    raw_bytes() = default;
+    raw_bytes(std::vector<unsigned char> b) : bytes(std::move(b)) {}
+
+    std::vector<unsigned char> bytes{};
+};
+
+// Comparison operator helper for the data types below
+template <class DataType>
+struct EqCompare {
```

Comment:
I'm a little puzzled about `EqCompare`. I see what it is trying to do (automatically define equality for each of `KinematicsData{Unspecified,SVA,URDF}`, but the way it does it raises concerns:

- It is an inherited, templated, free friend function of a template class. I do not know enough about C++ to reason about when such a thing is selected.
- It seems unnecessary. If `raw_bytes` just declared `inline friend bool operator==(const raw_bytes& lhs, const raw_bytes& rhs)`, wouldn't everything just work?